azerothcore/azerothcore-wotlk
# feat(Core/Player): add helper for send large count mail items

## Changes Proposed:
- Add new function `void SendItemRetrievalMail(std::vector<std::pair<uint32, uint32>> mailItems);` for send many items
- Add new function `void SendItemRetrievalMail(std::vector<std::pair<uint32, uint32>> mailItems);` for send one item

## Issues Addressed:
- Part #8431

## SOURCE:


## Tests Performed:
![image](https://user-images.githubusercontent.com/18329778/137024518-09f4946d-6370-46da-9675-c433cfc0d6e5.png)
![image](https://user-images.githubusercontent.com/18329778/137024535-af4d2164-d8c9-4914-a985-263587d39a9d.png)

## How to Test the Changes:
1. Add module in core https://github.com/WarheadCore/mod-test-mail
2. Use chat command `.testmailitem`
3. Check mailbox
4. Module send you next items in 2 functions:
```
1. item 40265 with count 15
2. item 40265 with count 5
3. item 49426 with count 9000
4. item 44615 with count 50
```

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
